backend/src/services/auto_trader.py

```
@@ -33,238 +23,448 @@ def __init__(self, socketio=None, data_fetcher_service=None, ai_analysis_service
+                    logger.warning(f"AutoTrader: Creator has full control for {token_address}")
+                    return False
+
+            # Check for suspicious patterns in token metadata/details
+            token_details = await self.data_fetcher_service.get_token_by_address(token_address)
+            if token_details:
+                if token_details.get('dev_wallet_active'):
+                    logger.warning(f"AutoTrader: Dev wallet still active/controlling for {token_address}")
 
+            logger.info(f"AutoTrader: Contract risk analysis passed for {token_address}")
+            return True
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)